docs/advanced-topics/dockstore-cli/dockstore-cli-faq.rst

```diff
@@ -0,0 +1,91 @@
+
+How do I launch tools/workflows without internet access on compute nodes?
+-------------------------------------------------------------------------
+
+Some tools/workflows require Docker images to launch even if they are
+local entries. If the compute nodes do not have internet access, you can
+follow these steps:
+
+1. download the Docker image(s) on the head node which does have internet access using the ``docker save -o <filename> <imagename>``
+2. ensure that the ``<imagename>`` matches the image name specific in the CWL or WDL descriptor 
+3. place the image file(s) in a location that the compute nodes have access to (make sure there are only images in that directory)
+4. specify in the dockstore config file (default ~/.dockstore/config) the directory that contains your image(s) using ``docker-images = /home/user/docker_images_directory``
+
+The Dockstore CLI will automatically load all Docker images in the
+directory specified prior to a ``launch --local-entry`` command.
+
+.. _cromwell-docker-lockup:
+
+I was running a WDL locally, but some of my tasks are failing randomly and/or now I cannot use Docker.
+------------------------------------------------------------------------------------------------------
+
+This is a known issue with how Cromwell and Docker, which the Dockstore CLI use to launch WDL workflows, manage resources on a local machine. Certain problems related to resource management may happen when running locally that do not happen when running on the cloud. These problems are much more likely to happen if you are running a computationally intensive scattered task, such as LD pruning 23 chromosomes where each chromosome is an instance of a scattered task. The two most common problems we see are a "Docker lockup" and stochastic failure of tasks. It is possible for one, both, or neither of these problems to occur during a single submission.
+
+If a Docker lockup happens, you will notice tasks do not progress beyond WaitingForReturnCode and you will be temporarily unable to "spin up" any Docker containers, even outside of Cromwell. Thankfully, this state can be resolved by restarting the Docker service via the Docker Desktop dropdown, or entering ``service docker restart`` on the command line.
+
+The other issue we often see is some instances of scattered tasks getting `sigkilled <https://www.gnu.org/software/libc/manual/html_node/Termination-Signals.html>`__ by the operating system. You will know when this happens because the `rc` (return code) file will read 137. If it reads anything except 137, then you can assume that it wasn't actually a resource management error and look in stderr or stdout for the true culprit.
```

Comment:
I thought I covered it by saying...
>  You will know when this happens because the `rc` (return code) file will read 137.
...but upon reading it again, maybe it should be clearer. Every task executed in Cromwell (including every instance of a scattered task) returns an actual file which contains nothing except the numeric return code. The problem is I don't know exactly where that file is for most users because I am affected by https://github.com/dockstore/dockstore/issues/4755

Reply:
> I thought I covered it by saying...
> 
> > You will know when this happens because the `rc` (return code) file will read 137.
> 

Whoops, I skimmed over that.

Reply:
Added an FAQ question just above it and linked the new entry to the Docker lockup entry.

**How do I find the return code for a WDL task?**
The numeric return code for a WDL task will be in that task’s execution folder. It is a single file named rc with no extension. Generally speaking, a 0 is a success, and anything else is a failure.

Let’s say you are running the vcf2gds workflow, which runs the check-gds task as a scattered task on an array of three files. Cromwell will refer to each instance of that scattered task as a “shard” and will name them starting with 0. If you notice that shard 0 seems to have failed, look for /cromwell-executions/[workflow ID]/call-check_gds/shard-0/execution/rc keeping in mind that the workflow ID will usually be a long mix of numbers, letters, and dashes such as 18a85cc0-aa59-4749-b1b9-e2580ed5e557.